en/net/advanced-operations/securing-and-signing/sign-pdf-document-from-smart-card/_index.md

```diff
@@ -184,26 +215,50 @@ It will be chosen automatically if you do not explicitly specify the hashing alg
 The ExternalSignature constructor also accepts a key certificate (it can be in Base64 format). You can pass a certificate containing a private key and a certificate containing only a public key. In either case, the signature will be performed externally in the CustomSignHash delegate code, but the external algorithm must create a signature corresponding to the key of the passed certificate. The certificate is needed to generate the signed document correctly. ECDSA signature does not support SHA-1.
 
 ```csharp
-var inputPdf = "";
-var outputPdf = "";
-
-using (var sign = new PdfFileSignature())
-{
-    sign.BindPdf(inputPdf);
+// For complete examples and data files, check for https://github.com/aspose-pdf/Aspose.PDF-for-.NET
 
-    X509Certificate2 signerCert = GetPublicCertificate();
+private static void SignWithExternalService()
+{    
+    // The path to the documents directory
+    string dataDir = RunExamples.GetDataDir_AsposePdf_SecuritySignatures();
     
-    var signature = new ExternalSignature(signerCert, DigestHashAlgorithm.Sha256);
-
-    SignHash customSignHash = delegate(byte[] signableHash, DigestHashAlgorithm digestHashAlgorithm)
+    // Open the document
+    using (var document = new Aspose.Pdf.Document(dataDir + "blank.pdf"))
     {
-        return CallExternalSignatureService(signableHash, digestHashAlgorithm);
-    };
-    
-    signature.CustomSignHash = customSignHash;
-    sign.Sign(1, "reason", "cont", "loc", false, new System.Drawing.Rectangle(0, 0, 500, 500), signature);
-    // Save result file
-    sign.Save(outputPdf);
+        using (var sign = new Aspose.Pdf.Facades.PdfFileSignature())
+        {
+            sign.BindPdf(document);
+            
+            // Get public certificate
+            X509Certificate2 signerCert = GetPublicCertificate();
```

Comment:
We may not specify full qualified name for the System libraries. However, `System.Security.Cryptography.X509Certificates` is enough hard case to specify it.